.github/workflows/environments-manifest.yml

```diff
@@ -0,0 +1,53 @@
+name: "Environments version manifest"
+
+# Gets the following content from the `environments.yml` manifest:
+# 1. [ENV]_WORDPRESS: the target version for the Terraform plan and apply on Staging.
+# 2. PREV_[ENV]_WORDPRESS: the previously deployed Terraform version.
+
+on:
+  workflow_call:
+    outputs:
+      STAG_WORDPRESS:
+        description: "The current version of the Staging Wordpress container"
+        value: ${{ jobs.environments-manifest.outputs.STAG_WORDPRESS }}
+      PREV_STAG_WORDPRESS:
+        description: "The previous version of the Staging Wordpress container"
+        value: ${{ jobs.environments-manifest.outputs.PREV_STAG_WORDPRESS }}
+      PROD_WORDPRESS:
+        description: "The current version of the Production Wordpress container"
+        value: ${{ jobs.environments-manifest.outputs.PROD_WORDPRESS }}
+      PREV_PROD_WORDPRESS:
+        description: "The previous version of the Production Wordpress container"
+        value: ${{ jobs.environments-manifest.outputs.PREV_PROD_WORDPRESS }}
+
+jobs:
+  environments-manifest:
+    runs-on: ubuntu-latest
+    outputs:
+      PROD_WORDPRESS: ${{ steps.versions.outputs.PROD_WORDPRESS }}
+      STAG_WORDPRESS: ${{ steps.versions.outputs.STAG_WORDPRESS }}
+      PREV_PROD_WORDPRESS: ${{ steps.previous.outputs.PROD_WORDPRESS }}
+      PREV_STAG_WORDPRESS: ${{ steps.previous.outputs.STAG_WORDPRESS }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Versions
+        id: versions
+        run: |
+          PROD_WORDPRESS=$(yq '.production.wordpress' < ./infrastructure/environments.yml)
+          STAG_WORDPRESS=$(yq '.production.apache' < ./infrastructure/environments.yml)
+          echo "$(cat ./infrastructure/environments.yml)"
+          echo "::set-output name=PROD_WORDPRESS::${PROD_WORDPRESS}"
+          echo "::set-output name=STAG_WORDPRESS::${STAG_WORDPRESS}"
+      - name: Previous versions
+        id: previous
+        run: |
+          PREV_SHA="$(git show -2 --pretty=format:"%h" --no-patch infrastructure/environments.yml | tail -n 1)"
```

Comment:
A risk this has for PRs to Staging is that this only gets the previous commit in the PR branch, so if `infrastructure/environments.yml` is updated multiple times in a single PR, you'll get inconsistent behaviour.

A way around this would be to compare the PR branch version of `infrastructure/environments.yml` to whatever is in `main`.